### PR TITLE
[SPARK-15668] [ML] ml.feature: update check schema to avoid confusion when user use MLlib.vector as input type

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

ml.feature: update check schema to avoid confusion when user use MLlib.vector as input type
## How was this patch tested?

existing ut

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala [hunk 1]
```diff
@@ -64,8 +64,7 @@ private[feature] trait MinMaxScalerParams extends Params with HasInputCol with H
   protected def validateAndTransformSchema(schema: StructType): StructType = {
     require($(min) < $(max), s"The specified min(${$(min)}) is larger or equal to max(${$(max)})")
     val inputType = schema($(inputCol)).dataType
```
Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala]: `inputType` is not required any longer, right?

File: mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala [hunk 2]
```diff
@@ -40,8 +40,7 @@ private[feature] trait MaxAbsScalerParams extends Params with HasInputCol with H
    /** Validates and transforms the input schema. */
   protected def validateAndTransformSchema(schema: StructType): StructType = {
     val inputType = schema($(inputCol)).dataType
```
Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala]: `inputType` is not required any longer, right?

**Comment by hhbyyh** [comment]
Thanks for helping review. @MLnick

**Comment by MLnick** [comment]
jenkins retest this please

File: mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala [hunk 3]
```diff
@@ -39,9 +39,7 @@ private[feature] trait MaxAbsScalerParams extends Params with HasInputCol with H
 
    /** Validates and transforms the input schema. */
   protected def validateAndTransformSchema(schema: StructType): StructType = {
-    val inputType = schema($(inputCol)).dataType
-    require(inputType.isInstanceOf[VectorUDT],
-      s"Input column ${$(inputCol)} must be a vector column")
+    SchemaUtils.checkColumnType(schema, $(inputCol), new VectorUDT)
```
Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala]: just a note on this - the fact that it requires `new VectorUDT` results in a message that contains `VectorUDT@XYZ` i.e. an instance, which is ok but not ideal. For the built-in types we have a case object that makes it cleaner, so we could think about doing that for VectorUDT as it is used a lot, e.g. 
```
private[spark] case object VectorUDT extends VectorUDT
```
Or alternatively, in `SchemaUtils.checkColumnType` we could use `getClass.getName` instead.

**Comment by MLnick** [comment]
I'm going to go ahead and merge this. We can make a call on  https://github.com/apache/spark/pull/13411#discussion_r65631481 and if anything needs doing we can do that in a follow up.

**Comment by MLnick** [comment]
Thanks, merged to master/branch-2.0

**Comment by MLnick** [comment]
Created https://issues.apache.org/jira/browse/SPARK-15746 to track the `VectorUDT@XYZ` error message printing.

**Pull request closed without merging**